pkg/utils/release/controller.go

```diff
@@ -109,29 +114,146 @@ func (s *SuiteController) GetPipelineRunInNamespace(namespace, releaseName, rele
 	return nil, err
 }
 
-// GetRelease returns the release with the given name in the given namespace.
-func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1alpha1.Release, error) {
-	release := &v1alpha1.Release{}
+// CreateReleasePlan creates a new ReleasePlan using the given parameters.
+func (s *SuiteController) CreateReleasePlan(name, source_namespace, application, target_namespace, AutoReleaseLabel string) (*appstudiov1alpha1.ReleasePlan, error) {
+	var releasePlan *appstudiov1alpha1.ReleasePlan
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlan is not missing
+		releasePlan = &appstudiov1alpha1.ReleasePlan{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: name,
+				Name:         name,
+				Namespace:    source_namespace,
+				Labels: map[string]string{
+					appstudiov1alpha1.AutoReleaseLabel: AutoReleaseLabel,
+				},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanSpec{
+				DisplayName: name,
+				Application: application,
+				Target: kcp.NamespaceReference{
+					Namespace: target_namespace,
+				},
+			},
+		}
+	} else { // AutoReleaseLabel in ReleasePlanAdmission is missing
+		releasePlan = &appstudiov1alpha1.ReleasePlan{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: name,
+				Name:         name,
+				Namespace:    source_namespace,
+				Labels:       map[string]string{},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanSpec{
+				DisplayName: name,
+				Application: application,
+				Target: kcp.NamespaceReference{
+					Namespace: target_namespace,
+				},
+			},
+		}
+	}
+	return releasePlan, s.KubeRest().Create(context.TODO(), releasePlan)
+}
+
+// GetReleasePlan returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlan(releasePlanName, devNamespace string) (*appstudiov1alpha1.ReleasePlan, error) {
+	releasePlan := &appstudiov1alpha1.ReleasePlan{}
 
 	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
-		Name:      releaseName,
-		Namespace: releaseNamespace,
-	}, release)
+		Name:      releasePlanName,
+		Namespace: devNamespace,
+	}, releasePlan)
 
-	return release, err
+	return releasePlan, err
+}
+
+// DeleteReleasePlan deletes the releasePlan resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
```

Comment:
why an extra space to begin this line? Also line should end in a period

Reply:
1. I always put a space line gap between functions to make them readable.
2. Added the period at endofline.

Reply:
I meant that there is an extra space before the word `specify` that doesn't exist before the word `Optionally` on the line before it

Reply:
My mistake , fix it